mollie/api/client.py

```diff
@@ -179,25 +187,81 @@ def perform_http_call(self, http_method, path, data=None, params=None):
             url += '?' + querystring
             params = None
 
-        try:
-            response = requests.request(
-                http_method, url,
-                verify=True,
-                headers={
-                    'Accept': 'application/json',
-                    'Authorization': 'Bearer {api_key}'.format(api_key=self.api_key),
-                    'Content-Type': 'application/json',
-                    'User-Agent': self.user_agent,
-                    'X-Mollie-Client-Info': self.UNAME,
-                },
-                params=params,
-                data=data,
-                timeout=self.timeout,
-            )
-        except Exception as err:
-            raise RequestError('Unable to communicate with Mollie: {error}'.format(error=err))
+        if self.oauth:
```

Comment:
Code could look better when there are 2 separate private methods that are called from `perform_http_call`: _perform_http_call_apikey()` and _perform_http_call_oauth()`.

Reply:
Done,

This also made me add a `_format_request_data` for logic that is used by both `_perform_http_call_apikey` and `_perform_http_call_oauth`. DRY.